File: typescript/src/tools/organizations/getDetails.ts
import { OrganizationGetDetailsSchema } from "@/schema/tool-inputs";
import { getToolDefinition } from "@/tools/toolDefinitions";
import type { Context, Tool } from "@/tools/types";
import type { z } from "zod";

const schema = OrganizationGetDetailsSchema;

type Params = z.infer<typeof schema>;

export const getDetailsHandler = async (context: Context, _params: Params) => {
	const orgId = await context.getOrgID();

	const orgResult = await context.api.organizations().get({ orgId });

	if (!orgResult.success) {
		throw new Error(`Failed to get organization details: ${orgResult.error.message}`);
	}

	return {
		content: [{ type: "text", text: JSON.stringify(orgResult.data) }],
	};
};

const definition = getToolDefinition("organization-details-get");

const tool = (): Tool<typeof schema> => ({
	name: "organization-details-get",
	description: definition.description,
	schema,
	handler: getDetailsHandler,
});

export default tool;

File: python/tools/organizations/get_details.py
import json

from api.client import is_error, is_success
from schema.tool_inputs import OrganizationGetDetailsSchema
from tools.tool_definitions import get_tool_definition
from tools.types import Context, Tool, ToolResult


async def get_organization_details_handler(context: Context, _params: OrganizationGetDetailsSchema) -> ToolResult:
    org_id = await context.get_org_id()
    org_result = await context.api.organizations().get(org_id)

    if is_error(org_result):
        raise Exception(f"Failed to get organization details: {org_result.error}")

    assert is_success(org_result)

    return ToolResult(content=json.dumps(org_result.data.model_dump(mode="json")))


def get_organization_details_tool() -> Tool[OrganizationGetDetailsSchema]:
    definition = get_tool_definition("organization-details-get")

    return Tool(
        name="organization-details-get",
        description=definition["description"],
        schema=OrganizationGetDetailsSchema,
        handler=get_organization_details_handler,
    )
